### PR TITLE
Major changes to cc wallet

### PR DESCRIPTION
**Pull request by richardkiss**
Take a look at these changes, see what you think, give me any feedback. Some of the changes are quite substantial.

### CODE REVIEW

File: src/wallet/cc_wallet/cc_wallet_puzzles.py [hunk 1]
```diff
@@ -15,42 +16,23 @@
 from src.util.ints import uint64
 
 
-def cc_make_puzzle(innerpuzhash, core):
-    # Puzzle runs the core, but stores innerpuzhash commitment
-    puzstring = f"(r (c (q 0x{innerpuzhash}) ((c (q {core}) (a)))))"
-    result = Program.to(binutils.assemble(puzstring))
-    return result
+MOD = load_clvm("coloured_coins.clvm")
```
Comment (Yostra) [inline src/wallet/cc_wallet/cc_wallet_puzzles.py:19]: I think we'll have to add this to pyinstaller spec, similar to how src/util/english.txt is added currently.

File: src/wallet/cc_wallet/cc_info.py [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
 from src.wallet.cc_wallet.ccparent import CCParent
 
 
 @dataclass(frozen=True)
 @streamable
 class CCInfo(Streamable):
-    my_core: Optional[str]  # core is stored as the disassembled string
     parent_info: List[Tuple[bytes32, Optional[CCParent]]]  # {coin.name(): CCParent}
-    my_colour_name: Optional[str]
+    my_colour_name: str
```
Comment (Yostra) [inline src/wallet/cc_wallet/cc_info.py:15]: This has not changed but I think we should rename this variable to just "colour", "my_colour_name" is easy to confuse with wallet name that is given by user.

File: src/wallet/cc_wallet/cc_info.py [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
+
+    def puzzle_for_inner_puzzle(self, inner_puzzle: Program) -> Program:
+        genesis_id = bytes.fromhex(self.my_colour_name) if self.my_colour_name else 0
```
Comment (Yostra) [inline src/wallet/cc_wallet/cc_info.py:18]: When does this case happen ?(not self.my_colour_name)

File: src/wallet/cc_wallet/cc_info.py (hunk 2 above)
Comment (richardkiss) [reply src/wallet/cc_wallet/cc_info.py:15]: I agree, and standardized names should be used. I prefer the name `genesis_id` over `colour`.

File: src/wallet/cc_wallet/cc_info.py (hunk 3 above)
Comment (richardkiss) [reply src/wallet/cc_wallet/cc_info.py:18]: Might be left over from when we had `Optional`. I'd love to get rid of it. There still is a case where `my_colour_name` is not being set, and I'd love to get of that too.

File: src/wallet/cc_wallet/cc_info.py (hunk 2 above)
Comment (matt-o-how) [reply src/wallet/cc_wallet/cc_info.py:15]: my_colour_name is the nickname for the colour that shows up in the UI, it is not an essential piece of information.
By default it gets set to the genesis_id but can be changed, i.e to "MattCoin". 
I think clarifying that it is a nickname of some kind is important.

Comment (richardkiss) [reply src/wallet/cc_wallet/cc_info.py:15]: Probably should add `mod_hash` here.

**Pull request closed without merging**